cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java

```diff
@@ -356,14 +360,38 @@ public void error(Throwable cause) {
     }
   }
 
+  /**
+   * On an authorization-enabled cluster, if caching is enabled too, updates the cache in the
+   * {@link AuthorizationEnforcementService} with the privileges of the user running the program.
+   */
+  private void updateAuthCache() {
```

Comment:
I mean why on the start up of the service cannot do this? E.g. it can just get the current UGI and do a first fetch on start up?